### DISCUSSION TITLE
Rules for inlining env variablea

### DISCUSSION THREAD

**Discussion by burtek** [vercel/next.js#29706] [Help] [answered]
What are the rules for (not) inlining non-public (i.e. not prefixed with `NEXT_PUBLIC_`) environment variables in the build? I have a whole subdirectory with source files for managing DB access, called only server-side, but for some reason env variables there are empty. So are those env variables only inlined on getStaticProps etc functions? What are the rules here?

**Comment by ijjk** [comment, marked as answer]
Hi, only `NEXT_PUBLIC_` values will be inlined in bundles, other variables need to be exposed in the env whenever starting Next.js to be available. Related documentation is available [here](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser)

**Reply by burtek** [reply]
But that still does not answer my question, my server-side code can't see env variables set in vercel's project settings

**Reply by burtek** [reply]
Nevermind, I'm dumb - my env variables are disabled for Preview builds...
